catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/SourcePoller.java

```diff
@@ -106,13 +101,32 @@ public void cancel(CatalogFramework framework, Map properties) {
      * Start method for this poller, invoked by the CatalogFramework's blueprint when the framework
      * is registered/installed. No logic is executed except for logging the framework name.
      *
-     * @param framework
-     *            the catalog framework being started
-     * @param properties
-     *            unused, but required by blueprint
+     * @param framework  the catalog framework being started
+     * @param properties unused, but required by blueprint
      */
     public void start(CatalogFramework framework, Map properties) {
         LOGGER.debug("Framework started for [{}]", framework);
     }
 
+    public int getPeriod() {
+        return period;
+    }
+
+    public void setPeriod(int period) {
+        this.period = period;
+        Optional.ofNullable(handle)
+                .ifPresent((handle) -> handle.cancel(false));
+        handle = scheduler.scheduleAtFixedRate(runner, initialDelay, this.period, TimeUnit.SECONDS);
+    }
+
+    public int getInitialDelay() {
+        return initialDelay;
+    }
+
+    public void setInitialDelay(int initialDelay) {
```

Comment:
Why would a user ever want to set an initial delay? Wouldn't that make the source appear unavailable until the initial delay has passed?

Recommend removing this configuration and always using 0.